server/athenian/api/precompute/accounts.py

```diff
@@ -418,47 +404,99 @@ async def report_precompute_success():
-    bots: Collection[str],
+    precomputed: bool,
+    bots: frozenset[str] | set[str],
     prefixer: Prefixer,
+    meta_ids: tuple[int, ...],
     sdb: Database,
     mdb: Database,
     cache: Optional[aiomcache.Client],
+    log: logging.Logger,
 ) -> tuple[int, int]:
-    """Copy the existing teams from GitHub and create a new team with all the involved bots \
-    for the specified account.
+    """
+    Take care of everything related to the teams.
 
-    :return: Number of copied teams and the number of noticed bots.
+    Create the root team and the bots team if they are missing.
+    Update the bots team with new bots from mdb.
+    If not precomputed, copy the teams from GitHub. Otherwise, synchronize them if the feature \
+    flag is enabled.
     """
-    root_team_id = await _ensure_root_team(account, sdb)
-    _, num_teams = await copy_teams_as_needed(account, meta_ids, root_team_id, sdb, mdb, cache)
-    num_bots = await _ensure_bot_team(account, bots, root_team_id, prefixer, sdb, mdb)
+    try:
+        root_team_id = await _ensure_root_team(account, sdb)
+    except Exception as e:
+        log.warning("root team %d: %s: %s", account, type(e).__name__, e)
+        sentry_sdk.capture_exception(e)
+        return 0, 0
+    try:
+        num_bots = await _ensure_bot_team(account, bots, root_team_id, prefixer, sdb, mdb, log)
+    except Exception as e:
+        log.warning("bots %d: %s: %s", account, type(e).__name__, e)
+        sentry_sdk.capture_exception(e)
+        num_bots = 0
+        # no return
+    if not precomputed:
+        log.info("considering account %d as brand new, creating the teams", account)
+        try:
+            _, num_teams = await copy_teams_as_needed(
+                account, meta_ids, root_team_id, sdb, mdb, cache,
+            )
+        except Exception as e:
+            log.warning("teams %d: %s: %s", account, type(e).__name__, e)
+            sentry_sdk.capture_exception(e)
+            num_teams = 0
+            # no return
+    else:
+        num_teams = 0
     return num_teams, num_bots
 
 
 async def _ensure_bot_team(
     account: int,
-    local_bots: Collection[str],
+    bots: frozenset[str] | set[str],
     root_team_id: int,
     prefixer: Prefixer,
     sdb: Database,
     mdb: Database,
+    log: logging.Logger,
 ) -> int:
-    bot_team = await get_bots_team(account, sdb)
-    if bot_team is not None:
-        return len(bot_team[Team.members.name])
-
-    bot_ids = set(chain.from_iterable(prefixer.user_login_to_node.get(u, ()) for u in local_bots))
-    await sdb.execute(
-        insert(Team).values(
-            Team(name=Team.BOTS, owner_id=account, parent_id=root_team_id, members=sorted(bot_ids))
-            .create_defaults()
-            .explode(),
-        ),
-    )
-    return len(bot_ids)
+    async with sdb.connection() as sdb_conn:
+        async with sdb_conn.transaction():
+            old_team = await fetch_bots.team(account, sdb_conn)
```

Comment:
we're making this query twice now, right ?

Reply:
Twice. This one is made inside a transaction, so that we don't have any contradictions.